shell/src/main/java/alluxio/cli/docgen/MetricsDocGenerator.java

```diff
@@ -124,30 +139,8 @@ public static void generate(boolean validate) throws IOException {
       boolean hasDiff = false;
 
       for (String category : CATEGORIES) {
-        ProcessBuilder pb = new ProcessBuilder("diff",
-            PathUtils.concatPath(csvFolder, ("temp-" + category + "-metrics." + CSV_SUFFIX)),
-            PathUtils.concatPath(csvFolder, (category + "-metrics." + CSV_SUFFIX)));
-        String output = IOUtils.toString(pb.start().getInputStream(), StandardCharsets.UTF_8);
-        if (output.length() > 0) {
-          System.out.println(output);
-          hasDiff = true;
-          System.out.println("Metrics file " + category + "-metrics." + CSV_SUFFIX + " changed.");
-        }
-        File f = new File(csvFolder, category + "-metrics." + CSV_SUFFIX);
-        File f2 = new File(csvFolder, "temp-" + category + "-metrics." + CSV_SUFFIX);
-        f2.renameTo(f);
-
-        pb = new ProcessBuilder("diff",
-            PathUtils.concatPath(ymlFolder, ("temp-" + category + "-metrics." + YML_SUFFIX)),
-            PathUtils.concatPath(ymlFolder, (category + "-metrics." + YML_SUFFIX)));
-        output = IOUtils.toString(pb.start().getInputStream(), StandardCharsets.UTF_8);
-        if (output.length() > 0) {
-          hasDiff = true;
-          System.out.println("Metrics file " + category + "-metrics." + YML_SUFFIX + " changed.");
-        }
-        f = new File(ymlFolder, category + "-metrics." + YML_SUFFIX);
-        f2 = new File(ymlFolder, "temp-" + category + "-metrics." + YML_SUFFIX);
-        f2.renameTo(f);
+        hasDiff = compareFiles(csvFolder,CSV_SUFFIX,category,hasDiff);
+        hasDiff = compareFiles(ymlFolder,YML_SUFFIX,category,hasDiff);
```

Comment:
the yml check overwrites the result of the csv check here? set different output booleans for each and join them